transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollRecvByteAllocatorHandle.java

```diff
@@ -32,8 +32,13 @@ final void receivedRdHup() {
         receivedRdHup = true;
     }
 
+    final boolean isReceivedRdHup() {
+        return receivedRdHup;
+    }
+
     boolean maybeMoreDataToRead() {
-        return isEdgeTriggered && lastBytesRead() > 0;
+        // If EPOLLRDHUP has been received we must read until we get a read error.
+        return isEdgeTriggered && (lastBytesRead() > 0 || receivedRdHup);
```

Comment:
@Scottmitch WDYT ? I think receivedRdHup is false more often then lastBytesRead() <= 0, so I think its fine as it is. /cc @fenik17